react/components/organisms/listing/listing.js

```diff
@@ -176,30 +203,180 @@ export class Listing extends Component {
         return Object.values(this.state.filters).filter(f => Boolean(f));
     }
 
+    _onScrollViewLayout(event) {
+        if (this.state.searchLoaded) return;
+        this.scrollViewWidth = event.nativeEvent.layout.width;
+        this.setState({ searchLoaded: this.scrollViewWidth !== 0 && this.searchHeaderWidth !== 0 });
+    }
+
+    _onSearchHeaderViewLayout(event) {
+        if (this.state.searchLoaded) return;
+        this.searchHeaderWidth = event.nativeEvent.layout.width;
+        this.setState({ searchLoaded: this.scrollViewWidth !== 0 && this.searchHeaderWidth !== 0 });
+    }
+
+    _expandSearchBar() {
+        // set select text as transparent
+        this.setState({ placeholderColorAlpha: 0 });
+
+        // animate search bar width to maximum
+        // and decrease dropdown select width
+        Animated.timing(this.state.searchWidth, {
+            toValue: 1,
+            duration: this.props.expandAnimationDuration,
+            useNativeDriver: false,
+            easing: Easing.inOut(Easing.ease)
+        }).start(() => {
+            this.setState({ expanded: true });
+        });
+    }
+
+    _shrinkSearchBar() {
+        // set select text as visible
+        this.setState({ placeholderColorAlpha: 1 });
+
+        // animate search bar width to normal
+        // and increase dropdown select width
+        Animated.timing(this.state.searchWidth, {
+            toValue: 0,
+            duration: this.props.expandAnimationDuration,
+            useNativeDriver: false,
+            easing: Easing.inOut(Easing.ease)
+        }).start(() => {
+            this.setState({ expanded: false });
+        });
+    }
+
+    _isHorizontalLayout = () => this.props.searchingHeaderLayout === "horizontal";
+
     _style() {
         return [styles.listing, this.props.style];
     }
 
-    _renderSearch() {
-        if (!this.props.search) return;
-
-        return <Search style={styles.search} onValue={this.onSearch} />;
+    _searchingHeaderStyle() {
+        const isHorizontal = this._isHorizontalLayout();
+        const layoutStyle = isHorizontal

```

Comment:
this can be simplified with just using this
```suggestion
        const layoutStyle = styles[`searchingHeader${this.props.searchingHeaderLayout}` ] || {}
```

Reply:
Can't because the prop name is lower case, but I think it seems more clear with the ternary also.